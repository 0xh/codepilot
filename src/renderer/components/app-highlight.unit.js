import AppHighlight from './app-highlight'

describe('@components/app-highlight', () => {
  it('injects a highlighting span over matched text', () => {
    const element = shallowMount(AppHighlight, {
      context: {
        props: {
          body: 'this.$store.state.test',
          matcher: '$store.state',
          className: 'testClass'
        }
      }
    })

    expect(element.text()).toEqual('this.$store.state.test')
    expect(element.contains('span.testClass')).toEqual(true)
    expect(element.html()).toMatchSnapshot()
  })

  it('doesnâ€™t do anything if no match is found', () => {
    const element = shallowMount(AppHighlight, {
      context: {
        props: {
          body: 'this.$store.state.test',
          matcher: 'vue-router',
          className: 'testClass'
        }
      }
    })

    expect(element.text()).toEqual('this.$store.state.test')
    expect(element.contains('span.testClass')).toEqual(false)
    expect(element.html()).toMatchSnapshot()
  })

  const queries = ['this.state', '$store test', 'state-this']

  queries.forEach(matcher => {
    it('highlights all partial matches', () => {
      const element = shallowMount(AppHighlight, {
        context: {
          props: {
            body: 'this.$store.state.test',
            matcher,
            className: 'testClass'
          }
        }
      })

      expect(element.text()).toEqual('this.$store.state.test')
      expect(element.findAll('.testClass').length).toEqual(2)
      expect(element.html()).toMatchSnapshot()
    })
  })
})
