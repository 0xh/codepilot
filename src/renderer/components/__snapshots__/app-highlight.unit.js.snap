// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/app-highlight doesnâ€™t do anything if no match is found 1`] = `<span>this.$store.state.test</span>`;

exports[`@components/app-highlight highlights all partial matches 1`] = `<span><span class="testClass">this</span>.$store.<span class="testClass">state</span>.test</span>`;

exports[`@components/app-highlight highlights all partial matches 2`] = `<span>this.$<span class="testClass">store</span>.state.<span class="testClass">test</span></span>`;

exports[`@components/app-highlight highlights all partial matches 3`] = `<span><span class="testClass">this</span>.$store.<span class="testClass">state</span>.test</span>`;

exports[`@components/app-highlight injects a highlighting span over matched text 1`] = `<span>this.<span class="testClass">$store.state</span>.test</span>`;
